proto/cronos/query.proto

```diff
@@ -30,6 +30,11 @@ service Query {
         option (google.api.http).get = "/cronos/v1/params";
     }
 
+    // Params queries permissions for a specific address..
+    rpc Permissions(QueryPermissionsRequest) returns (QueryPermissionsResponse) {
+        option (google.api.http).get = "/cronos/v1/permissions";
+    }
```

Comment:
should this PR bump to "v2", as these extra or breaking changes won't be backported to release/v1.0.x ?

Reply:
will the next upgrade be v2.0.x or v1.1.x ?

Reply:
v2 if it's breaking?

Reply:
yeah that is my assumption but very few cosmos chain (us included) have been following this convention, probably because adding features usually involved breaking changes and want to keep the major version for big breaking changes. 

I'm okay to bump it to v2 (probably in a separate PR) but may need to come to a consensus before hand

cc @yihuang @calvinaco

Reply:
"very few haven't been following this convention" == besides Cronos which just freshly released v1, I can't think of other chains ? cosmos hub was v1 up to v7 now, crypto.org chain was v1 up to v4 now, evmos was v1 up to v10...

Reply:
yeah I've been referring to the module. 

chain-main proto files are still on v1, ethermint and cosmos sdk as well. 

cronos uses the same versioning for both the module and the chain

Reply:
oh ok, I guess proto files can remain v1, but Go mod can be bumped to v2?

Reply:
I can do it in a separate PR to keep this one clean

Reply:
https://github.com/crypto-org-chain/cronos/pull/829

Reply:
not sure if we need `proto-swagger-gen` to update swagger as well

Reply:
breakage error is due to some package
```
Failure: buf.build/googleapis/googleapis:80720a488c9a414bb8d4a9f811084989: does not exist
```

I got somehow the same error when I try to generate swagger docs

Reply:
just tried, seems work after `buf mod update`
```make proto-swagger-gen
Generating Protobuf Swagger
Generate cronos swagger files...
```

Reply:
```
Generating Protobuf Swagger
Generate cronos swagger files
./cronos/query.proto
Generate cosmos swagger files
./cosmos/app/v1alpha1/query.proto
Failure: buf.build/googleapis/googleapis:80720a488c9a414bb8d4a9f811084989: does not exist
make: *** [proto-swagger-gen] Error 1

```

anyways I managed to generate query.swagger.json under tmp-swagger-gen but seems that a lot of swagger.json is missing from the config

Reply:
strange, I see this diff after generate
```
modified:   client/docs/statik/statik.go
modified:   client/docs/swagger-ui/swagger.yaml
modified:   third_party/proto/buf.lock
```

Reply:
Do you use `make proto-swagger-gen` ?

Anyways, I'll create a separate ticket to fix the swagger docs.

in config.json I can see its referring to a lot of different path such as 

```
./tmp-swagger-gen/ethermint/evm/v1/query.swagger.json
./tmp-swagger-gen/ethermint/feemarket/v1/query.swagger.json
```

which cannot be found in the repository

since its not related to this PR, we can work on that separately

Reply:
oh nvm it seems that the script combine all of them in the swagger.yaml

```
swagger-combine ./client/docs/config.json -o ./client/docs/swagger-ui/swagger.yaml -f yaml --continueOnConflictingPaths true --includeDefinitions true

```

could you help me push the diff in this PR @mmsqe while I trying to fix my environment ?